exchanges/request/request.go

```diff
@@ -1,300 +1,152 @@
-		Jobs:                 make(chan Job, MaxRequestJobs),
-		timeoutRetryAttempts: TimeoutRetryAttempts,
-		timedLock:            timedmutex.NewTimedMutex(DefaultMutexLockTimeout),
+	if i.Path == "" {
+		return nil, errors.New("invalid path")
 	}
-}
-
-// IsValidMethod returns whether the supplied method is supported
-func IsValidMethod(method string) bool {
-	return common.StringDataCompareInsensitive(supportedMethods, method)
-}
 
-// IsValidCycle checks to see whether the current request cycle is valid or not
-func (r *Requester) IsValidCycle(auth bool) bool {
-	if auth {
-		if time.Since(r.Cycle) < r.AuthLimit.GetDuration() {
-			return true
-		}
-	} else {
-		if time.Since(r.Cycle) < r.UnauthLimit.GetDuration() {
-			return true
-		}
-	}
-
-	r.StartCycle()
-	return false
-}
-
-func (r *Requester) checkRequest(method, path string, body io.Reader, headers map[string]string) (*http.Request, error) {
-	req, err := http.NewRequest(method, path, body)
+	req, err := http.NewRequest(i.Method, i.Path, i.Body)
 	if err != nil {
 		return nil, err
 	}
 
-	for k, v := range headers {
+	for k, v := range i.Headers {
 		req.Header.Add(k, v)
 	}
 
-	if r.UserAgent != "" && req.Header.Get("User-Agent") == "" {
-		req.Header.Add("User-Agent", r.UserAgent)
+	if r.UserAgent != "" && req.Header.Get(userAgent) == "" {
+		req.Header.Add(userAgent, r.UserAgent)
 	}
 
 	return req, nil
 }
 
 // DoRequest performs a HTTP/HTTPS request with the supplied params
-func (r *Requester) DoRequest(req *http.Request, path string, body io.Reader, result interface{}, authRequest, verbose, httpDebug, httpRecord bool) error {
-	if verbose {
+func (r *Requester) DoRequest(req *http.Request, p *Item) error {
+	if p.Verbose {
 		log.Debugf(log.Global,
-			"%s exchange request path: %s requires rate limiter: %v",
+			"%s exchange request path: %s",
 			r.Name,
-			path,
-			r.RequiresRateLimiter())
+			p.Path)
 
 		for k, d := range req.Header {
-			log.Debugf(log.Global, "%s exchange request header [%s]: %s", r.Name, k, d)
+			log.Debugf(log.Global,
+				"%s exchange request header [%s]: %s",
+				r.Name,
+				k,
+				d)
 		}
 		log.Debugf(log.Global,
-			"%s exchange request type: %s", r.Name, req.Method)
-		log.Debugf(log.Global,
-			"%s exchange request body: %v", r.Name, body)
+			"%s exchange request type: %s",
+			r.Name,
+			req.Method)
+
+		if p.Body != nil {
+			log.Debugf(log.Global,
+				"%s exchange request body: %v",
+				r.Name,
+				p.Body)
+		}
 	}
 
 	var timeoutError error
 	for i := 0; i < r.timeoutRetryAttempts+1; i++ {
+		// Initiate a rate limit reservation and sleep on requested endpoint
+		err := r.InitiateRateLimit(p.Endpoint)
+		if err != nil {
+			return err
+		}
+
 		resp, err := r.HTTPClient.Do(req)
 		if err != nil {
 			if timeoutErr, ok := err.(net.Error); ok && timeoutErr.Timeout() {
-				if verbose {
-					log.Errorf(log.ExchangeSys, "%s request has timed-out retrying request, count %d",
+				if p.Verbose {
+					log.Errorf(log.ExchangeSys,
```

Comment:
We're switching between log.ExchangeSys and log.Global, I think it would be good to have a separate logger for the requester. Especially for context lacking messages like: https://github.com/thrasher-corp/gocryptotrader/pull/413/files#diff-d0bee293ec050b1dd3e914acdef7f6a6R47